packages/core/src/hooks.ts

```diff
@@ -104,7 +104,16 @@ export function getSelectedConnector(
     return values[getIndex(connector)]
   }
 
-  function useSelectedProvider<T extends BaseProvider = Web3Provider>(connector: Connector, network?: Networkish) {
+  /**
+   * @typeParam T - A type argument must only be provided if one or more of the connectors passed to
+   * getSelectedConnector is using `connector.customProvider`, in which case it must match every possible type of this
+   * property, over all connectors. If any connector is not using `connector.customProvider`, T must also include
```

Comment:
nit: the last sentence is unnecessary (here and below), as `T extends BaseProvider` ensures that any connectors not using `customProvider` still have a valid typing (despite not being narrowed).